chef/cookbooks/keystone/recipes/server.rb

```diff
@@ -545,6 +544,18 @@
     tenant_name tenant
     action :add_tenant
   end
+
+  ruby_block "saving id for default #{tenant} tenant" do
+    block do
+      # note that we know because of what's at the end of this recipe that the
+      # .openrc has been created, and that the openstack cli is available
+      tenant_id = %x[source /root/.openrc && openstack project show openstack | awk '/id/  { print $4 }'].chomp
+      if !tenant_id.empty? && node[:keystone][tenant_type][:tenant_id] != tenant_id
+        node[:keystone][tenant_type][:tenant_id] = tenant_id
+        node.save
+      end
```

Comment:
add an 

only_if { node[:keystone][tenant_type][:tenant_id].empty? }

to avoid the repeated query. the id is not going to change ever..